### PR TITLE
[client][python] Add support for consumer get_last_message_id

### LINKED ISSUE #15985: Add getLastMessageId feature to Python Client
**Is your feature request related to a problem? Please describe.**
The `getLastMessageId` feature is available in the C++ client, but not in the python client.

**Describe the solution you'd like**
It'd be great if we can add the necessary python bindings to call the C++ client code.

### PR DESCRIPTION
**Pull request by michaeljmarshall**
Fixes #15985

### Motivation

Add the `get_last_message_id` method to the python consumer. This method is already present in the C++ client.

### Modifications

* Add method call to `consumer.cc`
* Add test 

### Verifying this change

There is a test to verify this change.

### Does this pull request potentially affect one of the following parts:

It expands the python client library.

### Documentation
Docs will be auto-generated for this addition.

### CODE REVIEW

File: pulsar-client-cpp/include/pulsar/Consumer.h [hunk 1]
```diff
@@ -390,6 +390,11 @@ class PULSAR_PUBLIC Consumer {
      */
     bool isConnected() const;
 
+    /**
+     * Get the latest message id for the consumer's topic.
+     */
+    virtual void getLastMessageIdAsync(BrokerGetLastMessageIdCallback callback);
```
Comment (BewareMyPower) [inline pulsar-client-cpp/include/pulsar/Consumer.h:396]: `BrokerGetLastMessageIdCallback` is an internal callback definition defined in `ConsumerImpl.h`, you should define your own callback. See `include/pulsar/Reader.h` for how it defines the `getLastMessageIdAsync`. And there is no need to make it `virtual`.

Comment (michaeljmarshall) [reply pulsar-client-cpp/include/pulsar/Consumer.h:396]: Thank you for your guidance! I'll take a look at that.

File: pulsar-client-cpp/python/pulsar_test.py [hunk 2]
```diff
@@ -486,6 +486,23 @@ def test_consumer_is_connected(self):
...
+        client = Client(self.serviceUrl)
+        topic = "test_consumer_get_last_message_id"
+        sub = "sub"
+        consumer = client.subscribe(topic, sub)
+
+        producer = client.create_producer(topic)
+
+        for i in range(10):
+            producer.send(b"hello-%d" % i)
+            message_id = consumer.get_last_message_id()
```
Comment (eolivelli) [inline pulsar-client-cpp/python/pulsar_test.py:499]: we should add a test with a empty topic

possibly we should cover the case of a truncated topic, but I am not sure we have a API to truncate a topic in python. not a blocker

Comment (michaeljmarshall) [reply pulsar-client-cpp/python/pulsar_test.py:499]: Sure, I can cover that test too. I need to address the underlying C++ first, though.

**Comment by michaeljmarshall** [comment]
Looks like this is superseded by one or more PRs. I'll close this for now.
